repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -70,29 +75,41 @@ class Session(interpreter: Interpreter)
 
   def state: SessionState = _state
 
-  def history: IndexedSeq[Statement] = _history
+  def statements: mutable.Map[Int, Statement] = _statements
+
+  def execute(code: String): Int = {
+    val statementId = newStatementId.getAndIncrement()
+    _statements.synchronized {
```

Comment:
I'm wondering if we could use `ConcurrentHashMap`, then this synchronization block is not necessary, what do you think?